pkg/resource/deploy/plan_executor.go

```diff
@@ -245,15 +272,15 @@ func (pe *planExecutor) performDeletes(ctx context.Context, opts Options) result
 
 // handleSingleEvent handles a single source event. For all incoming events, it produces a chain that needs
 // to be executed and schedules the chain for execution.
-func (pe *planExecutor) handleSingleEvent(event SourceEvent) result.Result {
+func (pe *planExecutor) handleSingleEvent(updateTargetsOpt map[resource.URN]bool, event SourceEvent) result.Result {
```

Comment:
Nit - Any reason this argument is first?

Reply:
Yes.  With parameters, i tend to pass them based on their scope in the caller.  In this case, updateTargetsOpt has a 'wider' scope than the 'event'.  i.e. we compute and store hte updateTargets once in the caller, but then we loop through the events one at a time, causing them to have a narrower lifetime/scope.  

For the callee this means that the rightmost argumetns are the ones that tend to be changing, while the leftmost ones (like `pe *planExecuter` itself) are the ones that are more fixed/less-changing.

I find that's a natural way to pass things that feels good for a large variety of cases.